### PR TITLE
Fixes/stream size/v6

### PR DESCRIPTION
**Pull request by victorjulien**
Describe changes:
- improve stream_size handling
- add option to continue to fully inspect tls when encrypted comms started

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
- PR victorjulien-pcap: https://buildbot.openinfosecfoundation.org/builders/victorjulien-pcap/builds/135
- PR victorjulien: https://buildbot.openinfosecfoundation.org/builders/victorjulien/builds/137

### CODE REVIEW

File: src/detect-parse.c [hunk 1]
```diff
@@ -1722,6 +1725,15 @@ static int SigValidate(DetectEngineCtx *de_ctx, Signature *s)
                 }
                 sm = sm->next;
             }
+            /* if stream_size is in use, also inspect packets */
+            sm = s->init_data->smlists[DETECT_SM_LIST_MATCH];
```
Comment (regit) [inline src/detect-parse.c]: I don't manage to see how we check that there is a raw content match. I think I've missed it but it is better to ask.

File: src/app-layer-ssl.c [hunk 2]
```diff
@@ -2568,14 +2578,31 @@ void RegisterSSLParsers(void)
...
-            if (ConfGetBool("tls.no-reassemble", &ssl_config.no_reassemble) != 1)
-                ssl_config.no_reassemble = SSL_CONFIG_DEFAULT_NOREASSEMBLE;
+        ConfNode *enc_handle = ConfGetNode("app-layer.protocols.tls.encryption-handling");
+        if (enc_handle != NULL && enc_handle->val != NULL) {
+            SCLogDebug("have app-layer.protocols.tls.encryption-handling = %s", enc_handle->val);
+            if (strcmp(enc_handle->val, "full") == 0) {
+                ssl_config.encrypt_mode = SSL_CNF_ENC_HANDLE_FULL;
+            } else if (strcmp(enc_handle->val, "bypass") == 0) {
+                ssl_config.encrypt_mode = SSL_CNF_ENC_HANDLE_BYPASS;
+            } else if (strcmp(enc_handle->val, "default") == 0) {
```
[the remark this answers is not in the document]
Comment (jasonish) [reply src/app-layer-ssl.c]: Would `track` make sense?

[the remark this answers is not in the document]
Comment (victorjulien) [reply src/app-layer-ssl.c]: Well this case doesn't just track, it also inspects higher level keywords. It runs with a disabled 'raw content' inspection.

File: src/detect-parse.c (hunk 1 above)
Comment (victorjulien) [reply src/detect-parse.c]: Making this more clean in the next iteration.

Comment (jasonish) [reply src/detect-parse.c]: So `ignore` isn't the best term either. `default` is OK if well documented.

**Comment by victorjulien** [comment]
Replaced by #3388

**Pull request closed without merging**